zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -80,6 +81,14 @@ final class MixedAnalyzingCompiler(
     logInputs(log, javaSrcs.size, scalaSrcs.size, outputDirs)
     val isPickleJava = config.currentSetup.order == Mixed && config.incOptions.pipelining && javaSrcs.nonEmpty
 
+    val earlyOut = config.earlyOutput.flatMap(_.getSingleOutput.toOption)
+    val pickleWrite = earlyOut.toList.flatMap { out =>
+      val sbv = scalac.scalaInstance.version.take(4)
+      if (out.toString.endsWith(".jar") && !Files.exists(out))
+        scala.reflect.io.RootPath(out, writable = true).close() // creates an empty jar
+      List("-Ypickle-write", out.toString).filter(_ => sbv == "2.12" || sbv == "2.13")
```

Comment:
@dnw This version dependent logic looks a bit fragile. e.g. it will break for Scala "2.120" 👽